sympy/core/expr.py

```diff
@@ -299,11 +299,16 @@ def __ge__(self, other):
         n2 = _n2(self, other)
         if n2 is not None:
             return _sympify(n2 >= 0)
-        if self.is_extended_real or other.is_extended_real:
-            dif = self - other
-            if dif.is_extended_nonnegative is not None and \
-                    dif.is_extended_nonnegative is not dif.is_extended_negative:
-                return sympify(dif.is_extended_nonnegative)
+        if self.is_extended_real and other.is_extended_real:
+            if self.is_infinite and other.is_infinite:
+                if (self * other).is_extended_positive:  # inf < inf & -inf < -inf
+                    return S.true
+            if (self.is_infinite and self.is_extended_nonnegative) \
+                    or (other.is_infinite and other.is_extended_nonpositive):
+                return S.false
+            nneg = (self - other).is_extended_nonnegative
```

Comment:
I thought the line above would be the only one necessary. Why are the other checks for is_infinite needed?

Maybe it would be better to fix those cases in Add.is_extended_nonnegative rather than here.

Reply:
They are for issue #16582, since after #16666 `oo < Dummy(extended_real=True)` returns `False` while `Dummy(extended_real =True) > oo` is unevaluated in current master. I tried to cover these issues in this PR.

Reply:
Can we not fix that by fixing Add._eval_is_extended_positive?
```
In [10]: x = Symbol('x', extended_real=True)                                                                                      

In [11]: print((x-oo).is_extended_positive)                                                                                       
None
```
I think if we can make it work here then we can make it work there and that's better since it is all defined in one place.

Reply:
`oo-oo` results in `nan` and `nan.is_extended_positive` is `False`. That's why the results here. I don't think `Add._eval_is_extended_positive` should be changed as such.

Reply:
`nan.is_extended_positive` is None but I take your point for `__gt__`. That doesn't apply for `__le__` and `__ge__` though right?

Reply:
> That doesn't apply for `__le__` and `__ge__` though right?

The present code evaluates `self - other` for __le__ and __ge__ also. Since, `oo - oo` or `nan` gives `None`, the code needs to be modified here only.